apps/passport-client/src/dispatch.ts

```diff
@@ -515,6 +522,17 @@ async function loadAfterLogin(
     SemaphoreIdentityPCDTypeName
   )[0] as SemaphoreIdentityPCD;
 
+  if (
+    !validateAndLogStateErrors(
+      userResponse.value,
+      identityPCD.claim.identity,
+      pcds
+    )
+  ) {
+    update({ userInvalid: true });
+    return;
```

Comment:
@artwyman - rather than doing an `update` here, should I instead return something so that the update happens outside of the sync code? should the sync code ever realistically call `update`?

Reply:
I think sync code can and does call update().  There's a massive call to update() near the bottom of this function